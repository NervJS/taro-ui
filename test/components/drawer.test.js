import Nerv, { findDOMNode } from 'nervjs'
import { renderToString } from 'nerv-server'
import { Simulate, renderIntoDocument } from 'nerv-test-utils'
import AtDrawer from '../../.temp/components/drawer/index'

const delay = ms => new Promise(resolve => setTimeout(resolve, ms))

describe('AtDrawer Snap', () => {
  it('render AtDrawer -- props show', () => {
    const component = renderToString(<AtDrawer show />)
    expect(component).toMatchSnapshot()
  })

  it('render AtDrawer -- props mask', () => {
    const component = renderToString(<AtDrawer show mask={false} />)
    expect(component).toMatchSnapshot()
  })

  it('render AtDrawer -- props width', () => {
    const component = renderToString(<AtDrawer show width='50%' />)
    expect(component).toMatchSnapshot()
  })

  it('render AtDrawer -- props items', () => {
    const component = renderToString(
      <AtDrawer show items={['菜单1', '菜单2']} />
    )
    expect(component).toMatchSnapshot()
  })
})

describe('AtDrawer Event', () => {
  it('AtDrawer onItemClick & onClose', async () => {
    const onItemClick = jest.fn()
    const onClose = jest.fn()
    const component = renderIntoDocument(
      <AtDrawer
        show
        items={['菜单1', '菜单2']}
        onItemClick={onItemClick}
        onClose={onClose}
      />
    )
    await delay(300)
    const items = findDOMNode(component, 'at-drawer').querySelectorAll(
      '.at-list__item'
    )
    const item0 = items[0]
    Simulate.click(item0)
    process.nextTick(() => {
      expect(onItemClick).toBeCalled()
    })
    await delay(350)
    expect(onClose).toBeCalled()
  })

  it('AtDrawer item NO.0 & NO.1 click, onItemClick(index) index should be 0 and 1', async () => {
    const onItemClick = jest.fn()
    const component = renderIntoDocument(
      <AtDrawer show items={['菜单1', '菜单2']} onItemClick={onItemClick} />
    )
    await delay(300)
    const items = findDOMNode(component, 'at-drawer').querySelectorAll(
      '.at-list__item'
    )
    const item0 = items[0]
    const item1 = items[1]
    Simulate.click(item0)
    Simulate.click(item1)
    process.nextTick(() => {
      expect(onItemClick.mock.calls[0][0]).toBe(0)
      expect(onItemClick.mock.calls[1][0]).toBe(1)
    })
  })

  it('AtDrawer click mask onClose', async () => {
    const onClick = jest.fn()
    const component = renderIntoDocument(
      <AtDrawer show items={['菜单1', '菜单2']} onClose={onClick} />
    )
    await delay(300)
    const componentDom = findDOMNode(component, 'at-drawer')
    const mask = componentDom.querySelector('.at-drawer__mask')
    Simulate.click(mask)
    await delay(350)
    expect(onClick).toBeCalled()
  })
})
