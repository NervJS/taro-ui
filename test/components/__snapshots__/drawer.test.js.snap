// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AtDrawer Snap render AtDrawer -- props items 1`] = `"<div class=\\"at-drawer at-drawer--left\\"><div style=\\"display:block;opacity:0;\\" class=\\"at-drawer__mask\\"></div><div style=\\"width:;transition:all 195ms cubic-bezier(0.4, 0, 0.6, 1);\\" class=\\"at-drawer__content\\"><div class=\\"at-list\\"><div class=\\"at-list__item\\"><div class=\\"at-list__item-container\\"><div class=\\"at-list__item-content item-content\\"><div class=\\"item-content__info\\"><div class=\\"item-content__info-title\\">菜单1</div></div></div><div class=\\"at-list__item-extra item-extra\\">undefined<div class=\\"item-extra__icon\\"><span class=\\"taro-text at-icon item-extra__icon-arrow at-icon-chevron-right\\"></span></div></div></div></div><div class=\\"at-list__item\\"><div class=\\"at-list__item-container\\"><div class=\\"at-list__item-content item-content\\"><div class=\\"item-content__info\\"><div class=\\"item-content__info-title\\">菜单2</div></div></div><div class=\\"at-list__item-extra item-extra\\">undefined<div class=\\"item-extra__icon\\"><span class=\\"taro-text at-icon item-extra__icon-arrow at-icon-chevron-right\\"></span></div></div></div></div></div></div></div>"`;

exports[`AtDrawer Snap render AtDrawer -- props mask 1`] = `"<div class=\\"at-drawer at-drawer--left\\"><div style=\\"display:none;opacity:0;\\" class=\\"at-drawer__mask\\"></div><div style=\\"width:;transition:all 195ms cubic-bezier(0.4, 0, 0.6, 1);\\" class=\\"at-drawer__content\\"></div></div>"`;

exports[`AtDrawer Snap render AtDrawer -- props show 1`] = `"<div class=\\"at-drawer at-drawer--left\\"><div style=\\"display:block;opacity:0;\\" class=\\"at-drawer__mask\\"></div><div style=\\"width:;transition:all 195ms cubic-bezier(0.4, 0, 0.6, 1);\\" class=\\"at-drawer__content\\"></div></div>"`;

exports[`AtDrawer Snap render AtDrawer -- props width 1`] = `"<div class=\\"at-drawer at-drawer--left\\"><div style=\\"display:block;opacity:0;\\" class=\\"at-drawer__mask\\"></div><div style=\\"width:50%;transition:all 195ms cubic-bezier(0.4, 0, 0.6, 1);\\" class=\\"at-drawer__content\\"></div></div>"`;
